packages/flutter/lib/src/rendering/unconstrained_box.dart

```diff
@@ -0,0 +1,115 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'box.dart';
+import 'debug_overflow_indicator.dart';
+import 'object.dart';
+import 'shifted_box.dart';
+import 'stack.dart';
+
+/// Renders a box, imposing no constraints on its child, allowing the child to
+/// render at its "natural" size.
+///
+/// This allows a child to render at the size it would render if it were alone
+/// on an infinite canvas with no constraints. This container will then expand
+/// as much as it can within its own constraints and align the child based on
+/// [alignment].  If the container cannot expand enough to accommodate the
+/// entire child, the child will be clipped.
+///
+/// In debug mode, if the child overflows the container, a warning will be
+/// printed on the console, and black and yellow striped areas will appear where
+/// the overflow occurs.
+///
+/// See also:
+///
+///  * [ConstrainedBox] for a box which imposes constraints on its child.
+///  * [Container], a convenience widget that combines common painting,
+///    positioning, and sizing widgets.
+///  * [OverflowBox], a widget that imposes different constraints on its child
+///    than it gets from its parent, possibly allowing the child to overflow
+///    the parent.
```

Comment:
since this is a RenderObject, we should link to the RenderObjects here. (The Widget equivalent should link to the Widgets).

Useful classes to link to here would be `RenderConstrainedOverflowBox` and `RenderSizedOverflowBox`.

Might be worth having some text here and in those classes (and equivalent text in the relevant widgets) that talks about what the difference is between this widget (considers overflow to be a failure, clips) and those widgets (consider overflow to be expected, don't clip, but won't hit text correctly on the overflowed regions).